core/services/relay/evm/contract_transmitter.go

```diff
@@ -87,14 +100,19 @@ func (oc *contractTransmitter) Transmit(ctx context.Context, reportCtx ocrtypes.
 	}
 	rawReportCtx := evmutil.RawReportContext(reportCtx)
 
-	oc.lggr.Debugw("Transmitting report", "report", hex.EncodeToString(report), "rawReportCtx", rawReportCtx, "contractAddress", oc.contractAddress)
+	txMeta, err := oc.reportToEthTxMeta(report)
+	if err != nil {
+		oc.lggr.Warnw("failed to generate tx metadata for report", "err", err)
+	}
+
+	oc.lggr.Debugw("Transmitting report", "report", hex.EncodeToString(report), "rawReportCtx", rawReportCtx, "contractAddress", oc.contractAddress, "txMeta", txMeta)
```

Comment:
From what I've seen the `reportToEthTxMetaNoop` function is only used on tests. Instead of passing a function, would it be possible to refactor this and have the function as a member of `contractTransmitter` struct?

Reply:
The function is a member of `contractTransmitter`, being passed in the constructor so we could aid in specific functions for CCIP plugins.
`reportToEthTxMetaNoop` is just an alias instead of an inline function